### PR TITLE
[bug] 6461: ensure docket entry has an index...

### PR DESCRIPTION
**Pull request by kkoskelin**
...by means of being added to a case entity's docket record it acquires an index

It may be that the `sendServedPartiesEmails` should be refactored to take a `docketEntryIndex` instead of a `docketEntryEntity`, which would allow the function to access `caseEntity.docketEntries[docketEntryIndex]` ?  

Open to discussion.

### CODE REVIEW

File: shared/src/business/useCases/docketEntry/updateDocketEntryInteractor.js [hunk 1]
```diff
@@ -98,6 +98,8 @@ exports.updateDocketEntryInteractor = async ({
     { applicationContext },
   );
 
+  caseEntity.updateDocketEntry(docketEntryEntity);
```
Comment (matthopson) [inline shared/src/business/useCases/docketEntry/updateDocketEntryInteractor.js]: You may want this between setting the document as served and sending served party emails.

See line 75 on `shared/src/business/utilities/shouldGenerateDocketRecordIndex.js`

**Comment by matthopson** [comment]
Just some general feedback: We need to evaluate the served status of a document before we can correctly set the index (see `shouldGenerateDocketRecordIndex.js`). We need the index set before service. So I think the general flow we're looking for here is something like:

- Create / Load `DocketEntry`
- Get served parties
- Set as served (`DocketEntry.setAsServed(servedParties)`) - gets `servedAt`
- Add / Update `DocketEntry` on Case - If adding, it will try to set an index, if updating, it will try to set an index ONLY if `isOnDocketRecord` flag is set on docket record.
- Send served parties emails - the passed `DocketEntry` item should have the index from the Add / Update step.

You'll also probably need to ensure the case gets updated with the new docket info - but you will likely be able to do that AFTER service since you pass the `DocketEntry` to the service method (so I don't think it reads it from `caseDetail`) - and this is probably already being done in the code anyway. Just something to note.

File: shared/src/business/entities/cases/Case.js [hunk 2]
```diff
@@ -1124,8 +1124,6 @@ Case.prototype.updateDocketEntry = function (updatedDocketEntry) {
       docketEntry.docketEntryId === updatedDocketEntry.docketEntryId,
   );
 
-  if (foundDocketEntry) Object.assign(foundDocketEntry, updatedDocketEntry);
```
Comment (kkoskelin) [inline shared/src/business/entities/cases/Case.js:1127]: This already happens on line 1138 and appears redundant?

**Review by matthopson — APPROVED** [review]
So, I think we _might_ have a few redundant calls to `Case.updateDocketEntry()`, but those ops are not costly, and certainly aren't hurting anything. Do we want to go through those and clean that up, or get this merged and re-evaluate?

**Comment by kkoskelin** [comment]
> So, I think we _might_ have a few redundant calls to `Case.updateDocketEntry()`, but those ops are not costly, and certainly aren't hurting anything. Do we want to go through those and clean that up, or get this merged and re-evaluate?

I think you're right, Matt -- there may be a few redundant ones in here -- but they are ultimately NOT very costly, and maybe even save our bacon.  Things are working well right now, and the workflow (as far as I can see it) is now including the docket entry indexes.  I'm calling it good.

Thanks for all your help, @matthopson !

**Pull request merged**